src/main/java/org/example/jvspringbootfirstbook/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,164 @@
+                        BigDecimal.valueOf(orderItem.getQuantity())
+                )
+        );
+        return orderItem;
+    }
+
+    private Set<OrderItem> getOrderItemSet(ShoppingCart cart, Order emptyOrder) {
+        Set<CartItem> cartItems = cart.getCartItems();
+        return cartItems.stream()
+                .map(orderItemMapper::fromCarttoOrderItem)
+                .map(orderItem -> setOrderToItem(orderItem, emptyOrder))
+                .collect(Collectors.toSet());
+    }
+
+    private OrderItem setOrderToItem(OrderItem orderItem, Order emptyOrder) {
+        orderItem.setOrder(emptyOrder);
+        return orderItem;
+    }
+
+    @Override
+    public List<OrderDto> getHistory(User user, Pageable pageable) {
+        return orderRepository.findAllByUser(user, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> getFromOrder(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "We don't have order with this id: " + orderId)
+                );
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getFromOrder(Long orderId, Long id) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "We don't have order with this id: " + orderId)
+                );
+        Set<OrderItem> orderItems = order.getOrderItems();
+        OrderItem orderItem = orderItems.stream()
+                .filter(item -> item.getId().equals(id))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "We can't find order Item with this id"
+                        )
+                );
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Override
+    public OrderDto updateStatus(Long id) {
+        if (!orderRepository.findById(id).isPresent()) {
+            throw new EntityNotFoundException("We don't have a order");
+        }
+        Order order = orderRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("We don't have a order")
+                );
+        if (order.getStatus().equals(Status.PENDING)) {
+            order.setStatus(Status.DELIVERED);
+            return getDto(order);
+        } else if (order.getStatus().equals(Status.DELIVERED)) {
+            order.setStatus(Status.COMPLETED);
+            return getDto(order);
+        } else {
+            order.setDeleted(true);
+            return getDto(order);
+        }
+    }
+
+    private OrderDto getDto(Order order) {
```

Comment:
Could be renamed to 'saveAndMapToDto()' as it better describes what the method does.